dsgrid/cloud/s3_storage_interface.py

```diff
@@ -64,15 +67,19 @@ def check_valid_lock_file(self, path):
         if relative_path == Path("configs/.locks"):
             pass
         elif relative_path == Path("data/.locks"):
-            assert False, "Data locks are currently not supported."
+            pass
         else:
             DSGMakeLockError(
                 "Lock file path provided must have relative path of configs/.locks or data/.locks"
             )
         return True
 
-    def get_lock_files(self):
-        contents = self._s3_filesystem.path(".locks").glob(pattern="*")
+    def get_lock_files(self, relative_path=None):
+        contents = list(self._s3_filesystem.path(".locks").glob(pattern="*"))
+        if relative_path:
+            for path in contents:
+                if not path.relative_to(relative_path):
+                    contents.remove(path)
```

Comment:
This modifies the list that is being iterated over. Python will raise an exception. You have to collect the indices in this loop and then delete them in reverse order in a second loop.

Reply:
Like this?:

```
    def get_lock_files(self, relative_path=None):
        contents = list(self._s3_filesystem.path(".locks").glob(pattern="*"))
        if relative_path:
            relative_contents = []
            for path in contents:
                if path.relative_to(relative_path):
                    relative_contents.append(path)
            return relative_contents
        return contents
```